### PR TITLE
Fix alignment in CComandBuffer

### PR DESCRIPTION
**Pull request by Jupeyy**
Will require C11 or C++17

### CODE REVIEW

**Comment by def-** [comment]
What do you mean by "or"? We only use C++11 I think.

**Comment by Jupeyy** [comment]
Well idc what we use, one of them is required

**Comment by Jupeyy** [comment]
if we wont target a new macos version, i'll just disable the alignment there i think

**Comment by Learath2** [comment]
We can't target macOS 10.15, the SDL bug still remains.

How about falling back to posix_memalign on macOS? That should be available
earlier

**Comment by Jupeyy** [comment]
so to sum it up, we fix all target os individually, i love standartization xD

File: src/base/system.c [hunk 1]
```diff
@@ -271,6 +273,17 @@ void mem_zero(void *block, unsigned size)
...
+{
+#if defined(CONF_PLATFORM_MACOSX)
+	void *ret_ptr;
+	posix_memalign(&ret_ptr, alignment, size);
+	return ret_ptr;
+#else
+	return aligned_alloc(alignment, size);
+#endif
+}
+
```
Comment (heinrich5991) [inline src/base/system.c:286]: From the [documentation of `max_align_t`](https://en.cppreference.com/w/cpp/types/max_align_t):
> Pointers returned by allocation functions such as `std::malloc` are suitably aligned for any object, which means they are aligned at least as strictly as `std::max_align_t`.

Sounds like we don't need this function unless we want to do SIMD or similar.

Comment (Jupeyy) [reply src/base/system.c:286]: so its not wrong?, sounds good

Comment (heinrich5991) [reply src/base/system.c:286]: Well, we don't need to introduce a dependency on a new version of C or POSIX if the function we used before does the exact same thing.

Comment (Jupeyy) [reply src/base/system.c:286]: well doesnt matter anyway, i'll wait for learath's pr and we'll use that

**Comment by heinrich5991** [comment]
Superseded by #3068.

**Pull request closed without merging**